### PR TITLE
getStats method with timeout

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
Sometime getStats (getting list of file changed) for a particular commit takes eternity.
making getStats call to execute in time-bound manner (configurable time)

### CODE REVIEW

File: internal/Configuration.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package internal
+
+import "github.com/caarlos0/env"
+
+type Configuration struct {
+	CommitStatsTimeoutInSec int `env:"COMMIT_STATS_TIMEOUT_IN_SEC" envDefault:"2"`
+}
+
+func ParseConfiguration() (*Configuration, error) {
+	cfg := &Configuration{}
```
Comment (kartik-579) [inline internal/Configuration.go:10]: We can make this more intuitive like CommitTimeoutConfiguration

File: pkg/git/RepositoryManager.go [hunk 2]
```diff
@@ -268,7 +272,17 @@ func (impl RepositoryManagerImpl) getStats(commit *object.Commit) (object.FileSt
...
 	}()
-	fs, err := commit.Stats()
+
+	// create context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(impl.configuration.CommitStatsTimeoutInSec)*time.Second)
+	defer cancel()
+	fs, err := commit.StatsContext(ctx)
+
+	// for timeout error, return empty result without error
+	if err == context.DeadlineExceeded || err == object.ErrCanceled {
```
Comment (kartik-579) [inline pkg/git/RepositoryManager.go:282]: Will this condition be checked in parallel if the commitStats method stucks in infinite loop?

File: internal/Configuration.go (hunk 1 above)
Comment (manish-agrawal-ai) [reply internal/Configuration.go:10]: This configuration is generic. (Think it as adding more configurations in future in this. )

File: pkg/git/RepositoryManager.go (hunk 2 above)
Comment (manish-agrawal-ai) [reply pkg/git/RepositoryManager.go:282]: provided context is with timeout. 
so if context is timed out, in that case err would be thrown by StatsContext method

Comment (kartik-579) [reply pkg/git/RepositoryManager.go:282]: Missed one line, sorted

**Review by kartik-579 — APPROVED** [review]

**Pull request merged**